bench/tiered/push_pull.c

```diff
@@ -214,31 +234,29 @@ recover_validate(const char *home, uint32_t num_records, uint64_t file_size, int
     testutil_wiredtiger_open(opts, home, buf, NULL, &conn, true, true);
     testutil_check(conn->open_session(conn, NULL, NULL, &session));
 
-    // srand((unsigned long)getpid() + num_records);
-    srand((uint32_t)getpid() + num_records);
-    str_len = sizeof(data_str) / sizeof(data_str[0]);
-    for (i = 0; i < str_len - 1; i++)
-        data_str[i] = 'a' + (uint32_t)rand() % 26;
+    /* Seed the random number generator */
+    v = (uint32_t)getpid() + num_records + (2 * counter);
+    __wt_random_init_custom_seed(&rnd, v);
 
-    data_str[str_len - 1] = '\0';
+    fill_random_data();
 
     testutil_check(__wt_snprintf(rm_cmd, sizeof(rm_cmd), "rm -rf %s/cache-bucket/*", home));
     status = system(rm_cmd);
     if (status < 0)
         testutil_die(status, "system: %s", rm_cmd);
 
     testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
-    val_1_size = strlen(data_str);
+    val_1_size = strlen((char *)data_str) + 1;
 
     gettimeofday(&start, 0);
 
     for (i = 0; i < num_records; i++) {
         cursor->set_key(cursor, i + 1);
         testutil_check(cursor->search(cursor));
```

Comment:
You can replace these two statements by
```
   testutil_check(cursor->next(cursor));
   testutil_check(cursor->get_key(cursor, &key));     // key declared as uint64_t
   testutil_assert(key == i + 1);
```
The reason is that iterating via `next` is more lightweight than doing `search` each time, and we want the benchmark to focus on speed of the connection, and minimize cpu in other places.  Sorry, I should have caught this before.

After the loop is complete you should do:
```
   testutil_assert(cursor->next(cursor) == WT_NOTFOUND);
```
That ensures we didn't read too much data due to some bug.

Reply:
1ac8107d21a11c5cb634b31bcedca81374709d70